usd_qtpy/render_util/framing_camera.py

```diff
@@ -0,0 +1,270 @@
+    valid values are: default, proxy, render, guide.
+    """
+    bbox_cache = UsdGeom.BBoxCache(time,purpose_tokens)
+    stage_root = stage.GetPseudoRoot()
+    return bbox_cache.ComputeWorldBound(stage_root).GetBox()
+
+def calculate_camera_position(camera: UsdGeom.Camera, stage: Usd.Stage, bounds_min: Gf.Vec3d = None, 
+                              bounds_max: Gf.Vec3d = None, z_up: bool = None, distance: float = None) -> Gf.Vec3d:
+    """
+    Calculate the world position for the camera based off of the size of the stage and the camera attributes.
+    Bounds and distance can be calculated beforehand so 
+    """
+    # Convenience. Life is short.
+    if not bounds_min or not bounds_max:
+        boundingbox = get_stage_boundingbox(stage)
+        bounds_min, bounds_max = boundingbox.GetMin(), boundingbox.GetMax()
+    
+    # Explicit None checks for values that can be False or 0
+    if z_up is None:
+        z_up = (_stage_up(stage) == "Z")
+
+    if distance is None:
+        distance = calculate_stage_distance_to_camera(camera, stage, bounds_min, bounds_max, z_up)
+
+    # Suppose a scene with a cone,
+    #             ..
+    #          ../  |    /\
+    #  O O    /     |   /  \
+    # [cam]<| ------|  /I am\
+    #         \..   | / cone \
+    #            \..|/ fear me\
+    #               ^       
+    #               |- The focus plane will be here, at the closest depth of the bounds
+    #                  of the scene the camera is pointed at.
+    #
+    # The center of the camera will be positioned at the center of the vertical and horizontal axis, 
+    # (Y and X respectively, assuming y up)
+    # and positioned back along the  with the calculated frustrum-filling distance along the depth axis,
+    # (Z, assuming y up).
+    
+    centroid = (bounds_min + bounds_max) / 2
+
+    if z_up:
+        camera_position = Gf.Vec3d(centroid[0], bounds_min[1]-distance, centroid[2]) 
+    else:
+        camera_position = Gf.Vec3d(centroid[0], centroid[1], bounds_max[2] + distance)
+
+    return camera_position
+
+def calculate_stage_distance_to_camera(camera: UsdGeom.Camera, stage: Usd.Stage, 
+                                       bounds_min: Gf.Vec3d = None, bounds_max: Gf.Vec3d = None, 
+                                       z_up: bool = None, fit: float = 1) -> float:
+    """
+    Calculates a distance from the centroid of the stage that would allow a camera to frame it perfectly.
+    Returns distance in stage units.
+    """
+    # Convenience. Life is short.
+    if not bounds_min or not bounds_max:
+        boundingbox = get_stage_boundingbox(stage)
+        bounds_min, bounds_max = boundingbox.GetMin(), boundingbox.GetMax()
+    
+    if z_up is None:
+        z_up = (_stage_up(stage) == "Z")
+
+    focal_length = camera.GetFocalLengthAttr().Get()
+    hor_aperture = camera.GetHorizontalApertureAttr().Get()
+    ver_aperture = camera.GetVerticalApertureAttr().Get()
+
+    ver_idx = 2 if z_up else 1
```

Comment:
Let's be explicit - no need to abbreviate here?
```suggestion
    vertical_axis_index = 2 if z_up else 1
```

Reply:
Good one, my thinking was to try and keep consistent with the other attributes in the same scope.
I'll adjust this one since it's not all that clear.